### PR TITLE
chore: add sourcemap for ts

### PR DESCRIPTION
**Pull request by ZLY201**
**What kind of change does this PR introduce?** (check at least one)

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update
- [ ] Refactor
- [x] Build-related changes
- [ ] Other, please describe:

**Does this PR introduce a breaking change?** (check one)

- [ ] Yes
- [x] No

If yes, please describe the impact and migration path for existing applications:

**The PR fulfills these requirements:**

- [x] It's submitted to the `dev` branch, **not** the `main` branch
- [x] When resolving a specific issue, it's referenced in the PR's title (e.g. `fix #xxx[,#xxx]`, where "xxx" is the issue number)
- [x] All tests are passing
- [x] New/updated tests are included

If adding a **new feature**, the PR's description includes:
- [x] A convincing reason for adding this feature (to avoid wasting your time, it's best to open a suggestion issue first and wait for approval before working on it)

**Other information:**

### CODE REVIEW

**Comment by ZLY201** [comment]
How can I delete the useless sourcemaps for isolate? @noname0310 

image

**Comment by noname0310** [comment]
I don't think we need a source map for d.ts, so it's better to remove this, right?

File: tools/buildFactory.ts [hunk 1]
```diff
@@ -137,7 +139,9 @@ export function gulpFactory(
     cleanStream,
     merge([tsBuildResult.dts, jsBuildResult])
```
Comment (noname0310) [inline tools/buildFactory.ts:140]: This method creates a source map only for js files.
```typescript
    merge([
      tsBuildResult.dts
        .pipe(filter(['**/*.ts', '!**/*.macro.d.ts'])),
      jsBuildResult
        .pipe(filter(['**/*.js', '!**/*.macro.js']))
        .pipe(sourceMap ? sourcemaps.write('.') : tap(() => { /* */ }))
    ])
```

**Review by noname0310 — COMMENTED** [review]
If you modify the function of line 227 like this, the problem will be solved.
```typescript
function filterDependencies() {
  return gulp.src(`${output}/**/*`, { read: false, allowEmpty: true, base: '.' })
    .pipe(filter([
      '**/*.ts',
      '**/*.js',
      '**/*.js.map',
      ...dependencySolver.getIncludedDependencies().map((dep) => `!${dep}`),
      ...dependencySolver.getIncludedDependencies().map((dep) => `!${dep}.map`)
    ]))
    .pipe(clean());
}
```

File: tools/buildFactory.ts (hunk 1 above)
Comment (ZLY201) [reply tools/buildFactory.ts]: Thanks!

**Review by noname0310 — APPROVED** [review]
The absence of the mangle from the umd build also seems to need to be reflected in the title.

**Pull request merged**